transports/xmpp/src/main/java/org/mule/transport/xmpp/XmppMessageDispatcher.java

```diff
@@ -80,6 +80,12 @@ protected MuleMessage doSend(MuleEvent event) throws Exception
 
     protected void sendMessage(MuleEvent event) throws Exception
     {
+        // Handle session closing by the server
+        if (!xmppConnector.getXmppConnection().isConnected())
+        {
+            xmppConnector.getXmppConnection().connect();
```

Comment:
An exception will be thrown. If the remote server is down or rejects the connection, Smack's reconnection mechanism will kick in.